backend/api/views.py

```diff
@@ -327,6 +328,55 @@ def put(self, request, id):
 
 		return 	Response(serializer.data, status=status.HTTP_200_OK)
 
+class RecommendedExamView(APIView):
+	def post(self, request):
+		serializer = CreateRecommendedExamSerializer(data=request.data)
+		serializer.is_valid(raise_exception=True)
+
+		subject = serializer.data.get("subject")
+		print(subject)
+
+		user_subject = request.user.profile.subjects.get(subject=subject)
+
+		# Fetch questions the user hasn't answered
+		questions_wrong = [i.answer for i in user_subject.wrongAnswers.all()]
+		questions_correct = [i.answer for i in user_subject.correctAnswers.all()]
+		questions_wrong_id = [i.id for i in questions_wrong]
+		questions_correct_id = [i.id for i in questions_correct]
+		questions_unanswered = list(Question.objects.exclude(id__in=questions_correct_id).exclude(id__in=questions_wrong_id))
+		
+		# Only insert a certain amount of unasnwered questions in the exam
+		if len(questions_unanswered) > MAX_UNANSWERED_QUESTIONS_RECOMMENDED:
+			questions_unanswered = random.sample(questions_unanswered, QUESTION_PER_EXAM)
+
+		questions = []
+
+		# Check if there are enough wrong answers to fill the exam, if not insert correct answers
+		space_left = QUESTION_PER_EXAM - len(questions_unanswered)
+		if len(questions_wrong) >= space_left:
+			questions_wrong_selected = random.sample(questions_wrong, space_left)
+
+			questions += questions_wrong_selected
+			# Return a "perfect" exam
+		
+		else:
+			# There are not enough wrong and unanswered questions so when need to get some right answers
+			space_left = QUESTION_PER_EXAM - len(questions_unanswered) - user_subject.wrongAnswers.count
+
+			questions_right_selected = random.sample(questions_correct, space_left)
+			questions += questions_right_selected
+
```

Comment:
The conditional structure is not the best possible.
Let's assume we have 7 Unanswered question and only 2 wrong answers in our user model. Then we go straight into the second conditional statement, disregarding the two wrong answers.
We should first add the wrong answers to the queue and then only then add correct answers if need be.

So I purpose a nested conditional instead

Space left is defined well in the second statement, but we never add any of the incorrect answers to the queue in the second statement